core/cog-launcher.c

```diff
@@ -175,6 +186,14 @@ cog_launcher_dispose (GObject *object)
         g_clear_object (&launcher->shell);
     }
 
+    if (launcher->sigint_source > 0)
+        g_source_remove (launcher->sigint_source);
+    launcher->sigint_source = 0;
```

Comment:
These three lines could be simplified using `g_clear_handle_id()` as follows:

```c
g_clear_handle_id (&launcher->sigint_source, g_source_remove);
```

(Same for `sigterm_source` below.)